featuretools/computational_backends/calculate_feature_matrix.py

```diff
@@ -557,15 +573,22 @@ def parallel_calculate_chunks(cutoff_time, chunk_size, feature_set, approximate,
                              cutoff_df_time_var=cutoff_df_time_var,
                              target_time=target_time,
                              pass_columns=pass_columns,
-                             progress_bar=None)
+                             progress_bar=None,
+                             progress_callback=progress_callback)
 
         feature_matrix = []
         iterator = as_completed(_chunks).batches()
         for batch in iterator:
             results = client.gather(batch)
             for result in results:
                 feature_matrix.append(result)
+                previous_progress = progress_bar.n
                 progress_bar.update(result.shape[0])
+                if progress_callback is not None:
+                    update = (progress_bar.n - previous_progress) / progress_bar.total * 100
```

Comment:
these 4 lines are repeated from above and contain a fair amount of logic. can we refactor to have the logic in only one place?